### ISSUE TITLE
Handle multiple transitions for same precursor-product ion pair

### ISSUE THREAD

**Issue by shubhra-agrawal** [ElucidataInc/ElMaven#405]
There are times when there are multiple transitions present for the same precursor-product ion pair. At present, ElMaven randomly picks up a transition when creating an EIC. This is also partly responsible for issues in #170 implementation.

Either the intensities from different transitions should be summed up or user should have control over which transition is picked.

**Comment by shubhra-agrawal** [comment]
@chubukov You mentioned using seeMS and xcalibur and getting almost 2x intensity compared to El-Maven (https://github.com/ElucidataInc/ElMaven/issues/170#issuecomment-331669948). Could it be because they are summing up intensities for all transitions while El-Maven is picking just one?

**Comment by chubukov** [comment]
No, that was for a single scan.

**Comment by chubukov** [comment]
@shubhra-agrawal I did like the suggestion proposed during the phone call (I think it was @rish9511? I can't remember now) of creating multiple groups when searching from a targeted list and matching multiple scan types.

**Comment by Raghavdata** [comment]
@chubukov Yes, this was the suggestion. I think we would go ahead with this plan.

**Comment by sahil21** [comment]
@shubhra-agrawal @chubukov 
So there are two cases to handle here:
1. When only one transition is picked up from multiple transitions.
2. When multiple transitions are clubbed together. For example:
![image](https://user-images.githubusercontent.com/21240516/31503613-c6c1e692-af8d-11e7-8107-f2214d0c9bc0.png)

These cases are completely different scenarios. 
- The first case occurs when SRM id is linked to the compound.
- The second case occurs when no SRM id is linked to the compound.

See issue: #89 #406

**Comment by sahil21** [comment]
@chubukov As mentioned in the previous comment, there are two cases to be handled here.

1. EICs are merged from multiple transitions
2. Only one transition is picked up from multiple transitions

As decided in the previous comments/call, we will give the user an option to select transition from the list of transitions using the drop-down menu on the EIC widget.

We are able to resolve the first case. It was a tricky implementation as getEIC(precursorMz, productMz) 
function only returns a single EIC but here we are expecting it to return multiple EICs for same Q1/Q3 pair. This works fine for Compound database widget & m/z text search. For targeted MS2 we need to handle it separately. We will merge this branch once testing is done.

For the second case, we have some solution in mind which we will implement in the coming sprint. Code implementation for this will be completely different from the first case.

**Comment by chubukov** [comment]
@sahil21 sounds fine. I still haven't really used the workflow of linking compounds to srmIDs, so I'm afraid I can't really comment on the failure cases there.
